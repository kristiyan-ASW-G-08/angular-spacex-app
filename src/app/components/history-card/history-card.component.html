<div class="card">
  <header class="card-header">
    <p class="card-header-title">
      {{ story.title }}
    </p>
  </header>
  <div class="card-content">
    <div class="content">
      {{ story.details }}
      <br />
      <time [dateTime]="story.event_date_utc">
        {{ story.event_date_utc | date }}</time
      >
    </div>
  </div>
  <footer class="card-footer">
    <a href="{{ story.links.article }}" class="card-footer-item">Article</a>
    <a href="{{ story.links.wikipedia }}" class="card-footer-item">Wikipedia</a>
    <a href="{{ story.links.reddit }}" class="card-footer-item">Reddit</a>
  </footer>
</div>
