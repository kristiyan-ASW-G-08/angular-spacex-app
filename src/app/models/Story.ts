export default interface Story {
  title: string;
  event_date_utc: string;
  event_date_unix: number;
  flight_number: number;
  details: string;
  links: {
    reddit: null | string;
    article: null | string;
    wikipedia: null | string;
  };
}
