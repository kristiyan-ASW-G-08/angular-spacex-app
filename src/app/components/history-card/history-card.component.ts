import { Component, Input } from "@angular/core";
import Story from "src/app/models/Story";

@Component({
  selector: "app-history-card",
  templateUrl: "./history-card.component.html"
})
export class HistoryCardComponent {
  @Input() story: Story;
}
