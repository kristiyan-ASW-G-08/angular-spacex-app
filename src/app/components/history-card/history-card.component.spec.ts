import { render } from "@testing-library/angular";
import { HistoryCardComponent } from "./history-card.component";
import Story from "src/app/models/Story";

const story = {
  id: 1,
  title: "Falcon 1 Makes History",
  event_date_utc: "2008-09-28T23:15:00Z",
  event_date_unix: 1222643700,
  details:
    "Falcon 1 becomes the first privately developed liquid fuel rocket to reach Earth orbit.",
  links: {
    reddit: null,
    article: "http://www.spacex.com/news/2013/02/11/flight-4-launch-update-0",
    wikipedia: "https://en.wikipedia.org/wiki/Falcon_1"
  }
};
const storyMock = (story as unknown) as Story;
describe("HistoryCardComponent", () => {
  it("should render", async () => {
    const container = await render(HistoryCardComponent, {
      componentProperties: { story: storyMock }
    });

    //@ts-ignore
    expect(container).toMatchSnapshot();
  });
});
