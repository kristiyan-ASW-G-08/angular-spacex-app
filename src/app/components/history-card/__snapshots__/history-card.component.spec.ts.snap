// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryCardComponent should render 1`] = `
Object {
  "abort": [Function],
  "animationEnd": [Function],
  "animationIteration": [Function],
  "animationStart": [Function],
  "blur": [Function],
  "canPlay": [Function],
  "canPlayThrough": [Function],
  "change": [Function],
  "click": [Function],
  "compositionEnd": [Function],
  "compositionStart": [Function],
  "compositionUpdate": [Function],
  "container": <div
    id="root0"
  >
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <p
          class="card-header-title"
        >
           Falcon 1 Makes History 
        </p>
      </header>
      <div
        class="card-content"
      >
        <div
          class="content"
        >
           Falcon 1 becomes the first privately developed liquid fuel rocket to reach Earth orbit. 
          <br />
          <time
            datetime="2008-09-28T23:15:00Z"
          >
             Sep 29, 2008
          </time>
        </div>
      </div>
      <footer
        class="card-footer"
      >
        <a
          class="card-footer-item"
          href="http://www.spacex.com/news/2013/02/11/flight-4-launch-update-0"
        >
          Article
        </a>
        <a
          class="card-footer-item"
          href="https://en.wikipedia.org/wiki/Falcon_1"
        >
          Wikipedia
        </a>
        <a
          class="card-footer-item"
          href=""
        >
          Reddit
        </a>
      </footer>
    </div>
  </div>,
  "contextMenu": [Function],
  "copy": [Function],
  "cut": [Function],
  "dblClick": [Function],
  "debug": [Function],
  "debugElement": null,
  "detectChanges": [Function],
  "doubleClick": [Function],
  "drag": [Function],
  "dragEnd": [Function],
  "dragEnter": [Function],
  "dragExit": [Function],
  "dragLeave": [Function],
  "dragOver": [Function],
  "dragStart": [Function],
  "drop": [Function],
  "durationChange": [Function],
  "emptied": [Function],
  "encrypted": [Function],
  "ended": [Function],
  "error": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "fixture": <app-history-card
    story={[Function Object]}
>
    <div
        class="card"
      >
        <header
          class="card-header"
        >
          <p
            class="card-header-title"
          >
             Falcon 1 Makes History 
          </p>
        </header>
        <div
          class="card-content"
        >
          <div
            class="content"
          >
             Falcon 1 becomes the first privately developed liquid fuel rocket to reach Earth orbit. 
            <br />
            <time
              datetime="2008-09-28T23:15:00Z"
            >
               Sep 29, 2008
            </time>
          </div>
        </div>
        <footer
          class="card-footer"
        >
          <a
            class="card-footer-item"
            href="http://www.spacex.com/news/2013/02/11/flight-4-launch-update-0"
          >
            Article
          </a>
          <a
            class="card-footer-item"
            href="https://en.wikipedia.org/wiki/Falcon_1"
          >
            Wikipedia
          </a>
          <a
            class="card-footer-item"
            href=""
          >
            Reddit
          </a>
        </footer>
      </div>
</app-history-card>,
  "focus": [Function],
  "focusIn": [Function],
  "focusOut": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "gotPointerCapture": [Function],
  "input": [Function],
  "invalid": [Function],
  "keyDown": [Function],
  "keyPress": [Function],
  "keyUp": [Function],
  "load": [Function],
  "loadStart": [Function],
  "loadedData": [Function],
  "loadedMetadata": [Function],
  "lostPointerCapture": [Function],
  "mouseDown": [Function],
  "mouseEnter": [Function],
  "mouseLeave": [Function],
  "mouseMove": [Function],
  "mouseOut": [Function],
  "mouseOver": [Function],
  "mouseUp": [Function],
  "navigate": [Function],
  "paste": [Function],
  "pause": [Function],
  "play": [Function],
  "playing": [Function],
  "pointerCancel": [Function],
  "pointerDown": [Function],
  "pointerEnter": [Function],
  "pointerLeave": [Function],
  "pointerMove": [Function],
  "pointerOut": [Function],
  "pointerOver": [Function],
  "pointerUp": [Function],
  "progress": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rateChange": [Function],
  "rerender": [Function],
  "scroll": [Function],
  "seeked": [Function],
  "seeking": [Function],
  "select": [Function],
  "selectOptions": [Function],
  "stalled": [Function],
  "submit": [Function],
  "suspend": [Function],
  "timeUpdate": [Function],
  "touchCancel": [Function],
  "touchEnd": [Function],
  "touchMove": [Function],
  "touchStart": [Function],
  "transitionEnd": [Function],
  "type": [Function],
  "volumeChange": [Function],
  "waitForDomChange": [Function],
  "waitForElement": [Function],
  "waitForElementToBeRemoved": [Function],
  "waiting": [Function],
  "wheel": [Function],
}
`;
